app/controllers/rooms_controller.rb

```diff
@@ -0,0 +1,141 @@
+class RoomsController < ApplicationController
+  def show
+    p current_user.show_solutions.to_s
+    @setup = Setup.take
+    @week = @setup.weeks.find_by_number(params[:week_number])
+
+    @room = Room.find_by_id(params[:room_number])
+
+    @next_room = @room.next(current_user.id,@week.id)
+    @previous_room = @room.previous(current_user.id,@week.id)
+
+    @learning_objects = @room.learning_objects.order(id: :asc).distinct
+    @results = UserToLoRelation.get_results(current_user.id,@week.id)
+
+    @user = current_user
+  end
+
+  def eval
+    @setup = Setup.take
+    @room = Room.find(params[:room_number])
+    learning_objects = @room.learning_objects.all.distinct
+
+    weight_comment = 3
+
+    @value = @room.score_limit.to_f
+
+    @week=Week.find_by_id(@room.week_id)
+
+    if (@week.learning_objects.distinct.count % 10 == 0)
+      @all = (@week.learning_objects.distinct.count / 10).to_i
+    else
+      @all = (@week.learning_objects.distinct.count / 10).to_i + 1
+    end
+    @count = @week.rooms.where("user_id = ?", current_user.id).count
+
+    if @room.state.to_s != "used"
+      score1 = 0
+
+      learning_objects.each do |l|
```

Comment:
V pripade, ze prechadzas kazdy `learning_object` daneho tyzdna a zaroven aj kazdy `feedback` daneho `learning_object`u, dochadza k velkemu problemu s neefektivnostou (kedze v kazdom kroku prveho cyklu sa pre dany `learning_object` musia ziskat `feedback`y). Nazyva sa to tiez **N+1 problem**.

Riesenie je velmi elegantne -- staci pouzivat metodu `eager_load`, ktorou si pri nacitavani `learning_object`-ov zaroven nacitas aj vsetky potrebne `feedback`y.

V tomto pripade teda:
`learning_objects.eager_load(:feedbacks).each do |l|`

Velmi dobry zdroj k `eager_load`:
http://blog.bigbinary.com/2013/07/01/preload-vs-eager-load-vs-joins-vs-includes.html